src/Extension/Settings/Initializers/ProjectUuidInitializer.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/**
+ * Copyright © 2019-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerSdk\Sdk\Extension\Settings\Initializers;
+
+use SprykerSdk\Sdk\Core\Appplication\Dependency\ProjectSettingRepositoryInterface;
+use SprykerSdk\SdkContracts\Entity\SettingInterface;
+use SprykerSdk\SdkContracts\Setting\SettingInitializerInterface;
+
+class ProjectUuidInitializer implements SettingInitializerInterface
+{
+    /**
+     * @var \SprykerSdk\Sdk\Core\Appplication\Dependency\ProjectSettingRepositoryInterface
+     */
+    protected ProjectSettingRepositoryInterface $projectSettingRepository;
+
+    /**
+     * @param \SprykerSdk\Sdk\Core\Appplication\Dependency\ProjectSettingRepositoryInterface $projectSettingRepository
+     */
+    public function __construct(ProjectSettingRepositoryInterface $projectSettingRepository)
+    {
+        $this->projectSettingRepository = $projectSettingRepository;
+    }
+
+    /**
+     * @param \SprykerSdk\SdkContracts\Entity\SettingInterface $setting
+     *
+     * @return void
+     */
+    public function initialize(SettingInterface $setting): void
+    {
+        $existingSetting = $this->projectSettingRepository->findOneByPath($setting->getPath());
+
+        if ($existingSetting && $existingSetting->getValues()) {
+            $setting->setValues($existingSetting->getValues());
+        } else {
+            $setting->setValues($this->uuid());
+            $this->projectSettingRepository->save($setting);
+        }
+    }
+
+    /**
+     * @return string
+     */
+    protected function uuid(): string
```

Comment:
I think this method should be in the separated class

Reply:
why? it's related to this Initializer